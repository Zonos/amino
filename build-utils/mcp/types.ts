/**
 * MCP Documentation Types
 * This file contains TypeScript types for the MCP documentation extraction process.
 */

/**
 * Represents extracted JSDoc comment information
 */
export type JSDocComment = {
  /** Description part of the JSDoc comment (text before any tags) */
  description: string;
  /** Location of the comment in the source file */
  location: SourceLocation;
  /** JSDoc tags extracted from the comment */
  tags: JSDocTag[];
  /** The full text of the JSDoc comment */
  text: string;
};

/**
 * Represents a specific JSDoc tag like @param, @default, @example, etc.
 */
export type JSDocTag = {
  /** The tag name without the @ symbol */
  name: string;
  /** The text content associated with this tag */
  text: string;
  /** The parsed value of this tag (if applicable) */
  value?: unknown;
};

/**
 * Represents a source file location (for tracking where comments were found)
 */
export type SourceLocation = {
  /** End line number (0-based) */
  endLine: number;
  /** File path relative to the project root */
  filePath: string;
  /** Start line number (0-based) */
  startLine: number;
};

/**
 * Represents basic component metadata
 */
export type ComponentMetadata = {
  /** Path to the component's main file */
  filePath: string;
  /** Component ID (typically kebab-case version of name) */
  id: string;
  /** Component name */
  name: string;
  /** Path to the component directory */
  path: string;
};

/**
 * Represents a component and its associated JSDoc comments
 */
export type ComponentDocumentation = {
  /** Top-level component JSDoc comment */
  comment?: JSDocComment;
  /** Path to the component's main file */
  filePath?: string;
  /** Component ID (typically kebab-case version of name) */
  id: string;
  /** Component name */
  name: string;
  /** Path to the component directory */
  path: string;
};

/**
 * Represents property information from an interface definition
 */
export type PropertyDefinition = {
  /** Default value of the property from @default tag */
  defaultValue?: string;
  /** Description of the property from JSDoc comment */
  description: string;
  /** Property name */
  name: string;
  /** Whether the property is required or optional */
  required: boolean;
  /** Type of the property as a string */
  type: string;
};

/**
 * Represents an interface definition extracted from a component file
 */
export type InterfaceDefinition = {
  /** Description of the interface from JSDoc comment */
  description: string;
  /** List of interfaces that this interface extends */
  extends: string[];
  /** Interface name */
  name: string;
  /** Properties defined in the interface */
  properties: PropertyDefinition[];
};

/**
 * Represents a component and its associated interface
 */
export type ComponentWithInterface = ComponentMetadata & {
  /** The component's interface definition, or null if not found */
  interface: InterfaceDefinition | null;
};

/**
 * Configuration options for the JSDoc extractor
 */
export type JSDocExtractorOptions = {
  /** Directory paths to scan for components */
  componentDirs: string[];
  /** Directories to exclude from component scanning */
  excludeDirs?: string[];
  /** File patterns to include/exclude (glob patterns) */
  filePatterns?: string[];
  /** Whether to include components marked as @private */
  includePrivate?: boolean;
  /** Output directory for documentation files */
  outputDir: string;
  /** Whether to use verbose logging */
  verbose?: boolean;
};

/**
 * Configuration options for the JSON file generator
 */
export type JsonFileGeneratorOptions = {
  /** Output directory for documentation files */
  outputDir: string;
  /** Whether to use verbose logging */
  verbose?: boolean;
};
