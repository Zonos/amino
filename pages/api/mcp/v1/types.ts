/**
 * Shared types for MCP API endpoints
 */

/**
 * Basic component metadata used in index listings
 */
export type ComponentMetadata = {
  category?: string;
  description?: string;
  id: string;
  name: string;
  path?: string;
  tags?: string[];
};

/**
 * Detailed component documentation
 */
export type ComponentDocumentation = {
  [key: string]: unknown;
  category?: string;
  description?: string;
  examples?: ComponentExample[];
  id: string;
  name: string;
  path?: string;
  props?: Record<string, PropDocumentation>;
  tags?: string[];
};

/**
 * Documentation for a component prop
 */
export type PropDocumentation = {
  [key: string]: unknown;
  defaultValue?: unknown;
  description?: string;
  name: string;
  required?: boolean;
  type?: string;
};

/**
 * Example usage of a component
 */
export type ComponentExample = {
  [key: string]: unknown;
  code?: string;
  description?: string;
  name?: string;
};

/**
 * Response structure for the components list endpoint
 */
export type ComponentsResponse = {
  components: ComponentMetadata[];
  pagination?: {
    limit: number;
    next?: string;
    offset: number;
    total: number;
  };
};

/**
 * Response structure for the health check endpoint
 */
export type HealthResponse = {
  components?: {
    api: {
      message?: string;
      status: 'ok' | 'error';
    };
    documentation: {
      componentCount?: number;
      message?: string;
      status: 'ok' | 'error';
    };
  };
  status: 'ok' | 'degraded' | 'error';
  timestamp: string;
  version: string;
};

/**
 * Standardized error response
 */
export type ErrorResponse = {
  error: {
    code: string;
    details?: Record<string, unknown>;
    message: string;
  };
};
