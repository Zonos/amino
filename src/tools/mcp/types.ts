/**
 * Types for Model Context Protocol (MCP) implementation
 */

export type ComponentListingItem = {
  /**
   * Brief description of the component
   */
  description?: string;

  /**
   * Unique identifier for the component
   */
  id: string;

  /**
   * Display name of the component
   */
  name: string;

  /**
   * Path to the component file relative to project root
   */
  path: string;

  /**
   * Tags associated with the component
   */
  tags?: string[];
};

export type ComponentListing = {
  /**
   * List of available components
   */
  components: ComponentListingItem[];
};

export type PropDefinition = {
  /**
   * Default value for the prop if any
   */
  defaultValue?: string;

  /**
   * Description of the prop
   */
  description?: string;

  /**
   * Example usage of the prop
   */
  example?: string;

  /**
   * Name of the prop
   */
  name: string;

  /**
   * Whether the prop is required
   */
  required: boolean;

  /**
   * Type of the prop
   */
  type: string;
};

export type ComponentDetails = {
  /**
   * Full description of the component
   */
  description?: string;

  /**
   * Usage examples
   */
  examples?: string[];

  /**
   * Unique identifier for the component
   */
  id: string;

  /**
   * Name of the component
   */
  name: string;

  /**
   * Path to the component file
   */
  path: string;

  /**
   * Component props definition
   */
  props: PropDefinition[];

  /**
   * Related components
   */
  relatedComponents?: string[];

  /**
   * Tags associated with the component
   */
  tags?: string[];
};

export type MCPClientOptions = {
  /**
   * Base URL for MCP server
   */
  baseUrl: string;
};

export type MCPError = {
  /**
   * Error code
   */
  code?: string;

  /**
   * Error message
   */
  message: string;
};

export type MCPResponse<T> = {
  /**
   * Response data
   */
  data?: T;

  /**
   * Error information if request failed
   */
  error?: MCPError;
};

export type FindComponentOptions = {
  /**
   * Whether to use exact name matching
   * @default true
   */
  exact?: boolean;
};
