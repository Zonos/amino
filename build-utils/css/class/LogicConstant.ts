import type { ConstantCustomizedComment } from 'build-utils/css/constants/logic/types/LogicConstantType';
import { capitalize } from 'build-utils/css/utils/capitalize';
import { formatTS } from 'build-utils/css/utils/formatTS';
import { readFileSync, writeFileSync } from 'fs';
import path from 'path';

type FileToParse = {
  /** @info file path to load (Need to be relative path - start from root of the project) */
  filePath: string;
};

type ParsedFile = FileToParse & {
  generateJsDocsFunction: ConstantCustomizedComment;
  hasJSDocsComment: boolean;
  keyValueParsed: Record<string, string> | null;
  rootFolder: string;
};

/**
 * Parse `ts` file from given path that has all the logic to generate key/value
 * and consume and generate constant
 */
export class LogicConstant {
  private parsedFile: ParsedFile;

  constructor() {
    const rootFolder = process.cwd();
    this.parsedFile = {
      filePath: '',
      generateJsDocsFunction: null,
      hasJSDocsComment: false,
      keyValueParsed: null,
      rootFolder,
    };
  }

  private printError(msg: string) {
    console.error(`File error '${__filename}': ${msg}`);
  }

  /**
   * Read and parse content from file to processable format
   * @param {FileToParse}
   */
  async parse({ filePath }: FileToParse): Promise<boolean> {
    this.parsedFile.filePath = filePath;
    if (filePath) {
      const parsedFileValid = this.checkValidParsedFile();
      if (!parsedFileValid) {
        this.printError(`Failed to parse file ${filePath}`);
      }
      return parsedFileValid;
    }
    this.printError('File path provided is invalid');
    return false;
  }

  private getFileName() {
    const { filePath } = this.parsedFile;
    return filePath.replace(/.+\/_?(.+)\.ts/gi, '$1');
  }

  /**
   * Check if file content satisfy naming rule for file before parsed or not
   */
  private async checkValidParsedFile(): Promise<boolean> {
    const { filePath, rootFolder } = this.parsedFile;
    /** Extract file name out of file path (without _) */
    const fileName = this.getFileName();
    const pascalFileName = capitalize(fileName);

    try {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      const importResult: Record<string, any> = await import(
        `${rootFolder}/${filePath}`
      );

      /** Make sure function to generate content exist */
      const requiredGenerateContentFunction = `get${pascalFileName}ConstantKeyValuePairs`;
      if (!importResult[requiredGenerateContentFunction]) {
        throw Error(`${requiredGenerateContentFunction} function is required`);
      }

      this.parsedFile.keyValueParsed =
        importResult[requiredGenerateContentFunction]();

      /** Make sure function to generate JSDocs exist */
      const requiredGenerateCommentFunction = `get${pascalFileName}ConstantCustomizedComment`;

      if (!importResult[requiredGenerateCommentFunction]) {
        throw Error(`${requiredGenerateCommentFunction} function is required`);
      }
      this.parsedFile.generateJsDocsFunction =
        importResult[requiredGenerateCommentFunction];

      /** Make sure `hasJSDocsComment` exist */
      const requiredHasJSDocsFlag = `hasJSDocsComment`;
      if (importResult[requiredHasJSDocsFlag] === undefined) {
        throw Error(`'${requiredHasJSDocsFlag}' constant is required.`);
      }
      this.parsedFile.hasJSDocsComment = importResult[requiredHasJSDocsFlag];
    } catch (ex) {
      if (ex instanceof Error) {
        this.printError(ex.message);
        return false;
      }
    }
    return true;
  }

  generateConstantContent() {
    const { generateJsDocsFunction, hasJSDocsComment, keyValueParsed } =
      this.parsedFile;
    if (keyValueParsed) {
      const contents = Object.entries(keyValueParsed).map(([key, value]) =>
        [
          /** Add JSDocsComment by adding content get function `generateJsDocsFunction` on each loop */
          hasJSDocsComment && generateJsDocsFunction
            ? `/** ${generateJsDocsFunction({ key, value })} */`
            : '',
          `'${key}': '${value}',`,
        ]
          .filter(Boolean)
          .join('\n'),
      );
      return contents.join('\n');
    }
    return null;
  }

  async generateConstantFile() {
    const { rootFolder } = this.parsedFile;
    const constantContent = this.generateConstantContent();
    const fileName = this.getFileName();
    if (constantContent) {
      const fileContent = `
      export const ${fileName} = {
        ${constantContent}
      } as const;
    `;
      const formatedContent = await formatTS(fileContent);
      writeFileSync(
        `${rootFolder}/build-utils/css/constants/generated/_${fileName}.ts`,
        formatedContent,
      );
    }
  }

  /** Checking camel case for a text */
  static camelCaseChecking(text: string) {
    return !!text && /^[a-z]/.test(text) && !/[\W_]/.test(text);
  }

  /**
   *
   * @param fileName file name that would be generated.
   *
   * **Note**:
   * - Need to be camelCase
   * - No extension
   * @return
   * - `null` when fileName is invalid
   * - `string` when fileName is valid
   */
  static async _generateLogicConstFileContent(
    fileName: string,
  ): Promise<string> {
    if (!this.camelCaseChecking(fileName)) {
      throw Error(
        'File name needs to be in valid camelCase format with no extension and no underscore.',
      );
    }
    const rootFolder = process.cwd();
    /** Read the template content */
    const templateContent = readFileSync(
      path.resolve(
        `${rootFolder}/build-utils/css/class/template/logicConstant.tpl`,
      ),
      { encoding: 'utf-8' },
    );
    const newTemplateContent = templateContent
      .replace(/{{FileName}}/g, fileName)
      .replace(/{{CapitalizedFileName}}/g, capitalize(fileName));

    return formatTS(newTemplateContent);
  }

  /**
   *
   * @param fileName file name that would be generated.
   * File name:
   * - Need to be camelCase
   * - Have no extension
   * @note
   * - Generated file will be located at `build-utils/css/constants/logic` with file name `_${fileName}.ts`.
   * - It will do nothing if fileName is in wrong format has nothing
   */
  static async generateLogicConstantFile(fileName: string) {
    const fileContent = await this._generateLogicConstFileContent(fileName);
    const rootFolder = process.cwd();
    writeFileSync(
      `${rootFolder}/build-utils/css/constants/logic/_${fileName}.ts`,
      fileContent,
    );
  }

  /**
   *
   * @param content : file content that need to converted;
   * @returns
   */
  static async transformImportedConstant(content: string) {
    const matchedImportLines = content.matchAll(
      /import {\s+(\w+)\s+} from '(.+)';/gm,
    );
    const matchedArr = Array.from(matchedImportLines);

    /** Key value pair of extracted constants */
    const extractedConstants = matchedArr.map(([, variable, filePath]) => {
      const rootFolder = process.cwd();
      /** Read generated file content */
      const fileContent = readFileSync(
        path.resolve(`${rootFolder}/${filePath}.ts`),
        { encoding: 'utf-8' },
      );
      /** Extract only constant content */
      const matchAll = fileContent.matchAll(
        /(?<=.+ = {\n +)((\n|.)+)(?=}.+)/gm,
      );
      const matched = Array.from(matchAll);
      const result = matched
        .map(([extractedVariable]) => extractedVariable)
        .join('');
      return {
        data: result,
        key: variable,
      };
    });

    const result = content
      /** Clean up all imports */
      .replace(/import {\s+(\w+)\s+} from '(.+)';/gm, '')
      /** Replace spread constant in theme constant with extracted constant from generated constant file */
      .replace(
        /\.\.\.(\w+),/gm,
        (_g0, variable) =>
          extractedConstants.find(item => item.key === variable)?.data || _g0,
      );
    return formatTS(result);
  }
}
