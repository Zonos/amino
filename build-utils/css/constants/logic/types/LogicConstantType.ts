export type KeyValuePairType = {
  /** Key type from each item in `ConstantKeyValuePairsType` */
  key: string;
  /** Value type from each item im `ConstantKeyValuePairsType` */
  value: string;
};

/** @info Logic to generate customized comment for base on key/value of each constant */
export type ConstantCustomizedComment =
  | (({ key, value }: KeyValuePairType) => string | null)
  | null;

export type ConstantKeyValuePairsType = () => Record<
  KeyValuePairType['key'],
  KeyValuePairType['value']
>;
