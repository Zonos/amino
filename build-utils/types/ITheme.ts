export type ITheme = 'day' | 'night' | 'midnight';
