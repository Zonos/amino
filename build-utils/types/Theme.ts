export type Theme = 'day' | 'night' | 'midnight';
