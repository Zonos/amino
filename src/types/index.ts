export { Depth } from './Depth';
export type { Intent } from './Intent';
export { Surface } from './Surface';
export type { IAminoTheme } from './IAminoTheme';
export type { ConfirmDialogOpts } from './ConfirmDialogOpts';
export type { Color } from './Color';
export type { IconProps } from './IconProps';
export type { AlertDialogOpts } from './AlertDialogOpts';
