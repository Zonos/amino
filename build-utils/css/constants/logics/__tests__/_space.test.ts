import { LogicConstant } from '../../../class/LogicConstant';
import {
  getSpaceConstantCustomizedComment,
  getSpaceConstantKeyValuePairs,
} from '../_space';
import { KeyValuePairType } from '../types/LogicConstantType';

describe('Make sure logic to generate `space` work as expected', () => {
  test('Key value pairs return correctly', () => {
    expect(getSpaceConstantKeyValuePairs()).toMatchSnapshot();
  });
});

describe('Make sure logic of generating jsdocs comment work as expected', () => {
  type UnitTest = {
    input: KeyValuePairType;
    expected: string;
  };
  const testCases: UnitTest[] = [
    {
      input: {
        key: 'space-24',
        value: '1.5rem',
      },
      expected: '@info space-24: 1.5rem (24px)',
    },
    {
      input: {
        key: 'space-0',
        value: '0',
      },
      expected: '@info space-0: 0 (0)',
    },
    {
      input: {
        key: 'space-negative-4',
        value: '-0.25rem',
      },
      expected: '@info space-negative-4: -0.25rem (-4px)',
    },
  ];
  test.each(testCases)(
    'Make sure space constant comment works as expected',
    ({ input: { key, value }, expected }) => {
      if (getSpaceConstantCustomizedComment) {
        const result = getSpaceConstantCustomizedComment({ key, value });
        expect(result).toBe(expected);
      }
    }
  );
});

describe('Make sure constant content are generated correctly', () => {
  test('`generateConstantContent` function return correctly', async () => {
    const logic = new LogicConstant();
    await logic.parse({
      filePath: 'build-utils/css/constants/logics/_space.ts',
    });

    expect(logic.generateConstantContent()).toMatchSnapshot();
  });
});
