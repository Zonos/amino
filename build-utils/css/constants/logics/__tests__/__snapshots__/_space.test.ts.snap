// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make sure constant content are generated correctly \`generateConstantContent\` function return correctly 1`] = `
"/** @info space-negative-80: -5rem (-80px) */
'space-negative-80': '-5rem',
/** @info space-negative-76: -4.75rem (-76px) */
'space-negative-76': '-4.75rem',
/** @info space-negative-72: -4.5rem (-72px) */
'space-negative-72': '-4.5rem',
/** @info space-negative-68: -4.25rem (-68px) */
'space-negative-68': '-4.25rem',
/** @info space-negative-64: -4rem (-64px) */
'space-negative-64': '-4rem',
/** @info space-negative-60: -3.75rem (-60px) */
'space-negative-60': '-3.75rem',
/** @info space-negative-56: -3.5rem (-56px) */
'space-negative-56': '-3.5rem',
/** @info space-negative-52: -3.25rem (-52px) */
'space-negative-52': '-3.25rem',
/** @info space-negative-48: -3rem (-48px) */
'space-negative-48': '-3rem',
/** @info space-negative-44: -2.75rem (-44px) */
'space-negative-44': '-2.75rem',
/** @info space-negative-40: -2.5rem (-40px) */
'space-negative-40': '-2.5rem',
/** @info space-negative-36: -2.25rem (-36px) */
'space-negative-36': '-2.25rem',
/** @info space-negative-32: -2rem (-32px) */
'space-negative-32': '-2rem',
/** @info space-negative-28: -1.75rem (-28px) */
'space-negative-28': '-1.75rem',
/** @info space-negative-24: -1.5rem (-24px) */
'space-negative-24': '-1.5rem',
/** @info space-negative-20: -1.25rem (-20px) */
'space-negative-20': '-1.25rem',
/** @info space-negative-16: -1rem (-16px) */
'space-negative-16': '-1rem',
/** @info space-negative-12: -0.75rem (-12px) */
'space-negative-12': '-0.75rem',
/** @info space-negative-8: -0.5rem (-8px) */
'space-negative-8': '-0.5rem',
/** @info space-negative-4: -0.25rem (-4px) */
'space-negative-4': '-0.25rem',
/** @info space-0: 0 (0) */
'space-0': '0',
/** @info space-4: 0.25rem (4px) */
'space-4': '0.25rem',
/** @info space-8: 0.5rem (8px) */
'space-8': '0.5rem',
/** @info space-12: 0.75rem (12px) */
'space-12': '0.75rem',
/** @info space-16: 1rem (16px) */
'space-16': '1rem',
/** @info space-20: 1.25rem (20px) */
'space-20': '1.25rem',
/** @info space-24: 1.5rem (24px) */
'space-24': '1.5rem',
/** @info space-28: 1.75rem (28px) */
'space-28': '1.75rem',
/** @info space-32: 2rem (32px) */
'space-32': '2rem',
/** @info space-36: 2.25rem (36px) */
'space-36': '2.25rem',
/** @info space-40: 2.5rem (40px) */
'space-40': '2.5rem',
/** @info space-44: 2.75rem (44px) */
'space-44': '2.75rem',
/** @info space-48: 3rem (48px) */
'space-48': '3rem',
/** @info space-52: 3.25rem (52px) */
'space-52': '3.25rem',
/** @info space-56: 3.5rem (56px) */
'space-56': '3.5rem',
/** @info space-60: 3.75rem (60px) */
'space-60': '3.75rem',
/** @info space-64: 4rem (64px) */
'space-64': '4rem',
/** @info space-68: 4.25rem (68px) */
'space-68': '4.25rem',
/** @info space-72: 4.5rem (72px) */
'space-72': '4.5rem',
/** @info space-76: 4.75rem (76px) */
'space-76': '4.75rem',
/** @info space-80: 5rem (80px) */
'space-80': '5rem',"
`;

exports[`Make sure logic to generate \`space\` work as expected Key value pairs return correctly 1`] = `
Object {
  "space-0": "0",
  "space-12": "0.75rem",
  "space-16": "1rem",
  "space-20": "1.25rem",
  "space-24": "1.5rem",
  "space-28": "1.75rem",
  "space-32": "2rem",
  "space-36": "2.25rem",
  "space-4": "0.25rem",
  "space-40": "2.5rem",
  "space-44": "2.75rem",
  "space-48": "3rem",
  "space-52": "3.25rem",
  "space-56": "3.5rem",
  "space-60": "3.75rem",
  "space-64": "4rem",
  "space-68": "4.25rem",
  "space-72": "4.5rem",
  "space-76": "4.75rem",
  "space-8": "0.5rem",
  "space-80": "5rem",
  "space-negative-12": "-0.75rem",
  "space-negative-16": "-1rem",
  "space-negative-20": "-1.25rem",
  "space-negative-24": "-1.5rem",
  "space-negative-28": "-1.75rem",
  "space-negative-32": "-2rem",
  "space-negative-36": "-2.25rem",
  "space-negative-4": "-0.25rem",
  "space-negative-40": "-2.5rem",
  "space-negative-44": "-2.75rem",
  "space-negative-48": "-3rem",
  "space-negative-52": "-3.25rem",
  "space-negative-56": "-3.5rem",
  "space-negative-60": "-3.75rem",
  "space-negative-64": "-4rem",
  "space-negative-68": "-4.25rem",
  "space-negative-72": "-4.5rem",
  "space-negative-76": "-4.75rem",
  "space-negative-8": "-0.5rem",
  "space-negative-80": "-5rem",
}
`;
